src/bolt/utils.ts

```diff
@@ -59,55 +63,122 @@ export async function getReactionData (event: ReactionAddedEvent | ReactionRemov
+        if (message.text) {
+            const match = /https:\/\/github.com\/[^ ]+?\/pull\/\d+/.exec(message.text);
+            pullRequestLink = match ? match[0] : '';
+        }
+
+        // If there is no PR info, try to check the message attachments, message is coming from github event.
+        if (!pullRequestLink.length && message.attachments?.length) {
+            const {title} = message.attachments[0];
+            if (title) {
+                const match = /https:\/\/github.com\/[^ ]+?\/pull\/\d+/.exec(title);
+                pullRequestLink = match ? match[0] : '';
+            }
+        }
+
         return {
             slackMsgText: message.text,
             slackMsgId: message.client_msg_id,
             slackMsgUserId: message.user,
-            slackThreadTs: message.thread_ts ?? message.ts
+            slackThreadTs: message.thread_ts ?? message.ts,
+            pullRequestLink,
         } as ReactionData;
     };
     const botUserId = await getBotUserId();
-    let messageInfo = null;
+    let messageInfo = getMessageInfo(result);
+
+    if (!result.client_msg_id && botUserId === event.item_user && !messageInfo.pullRequestLink.length &&
+        messageInfo.slackMsgTs !== messageInfo.slackThreadTs) {
 
-    if (!result.client_msg_id && botUserId === event.item_user) {
-        messageInfo = getMessageInfo(result);
         result = await slackBotApp.client.conversations.replies({
             channel: event.item.channel,
             inclusive: true,
             latest: messageInfo.slackThreadTs,
             limit: 1,
             ts: messageInfo.slackThreadTs,
         });
-
+        messageInfo = getMessageInfo(result);
     }
 
-    messageInfo = getMessageInfo(result);
-
     result = await slackBotApp.client.chat.getPermalink({
         channel: event.item.channel,
         message_ts: messageInfo.slackThreadTs,
     });
 
     const {permalink} = result;
 
-    const match = /https:\/\/github.com\/[^ ]+?\/pull\/\d+/.exec(messageInfo.slackMsgText as string);
     return {
         ...messageInfo,
         slackPermalink: permalink as string,
         reaction: event.reaction,
         reactionUserId: event.user,
-        pullRequestLink: match ? match[0] : '',
         slackChannelId: event.item.channel,
     };
 }
 
+
+export async function postSlackMessage (slackMessage: ChatPostMessageArguments): Promise<void> {
+    await slackBotApp.client.chat.postMessage(slackMessage);
+}
+
+export async function sendCodeReviewSummary (channel: string): Promise<void> {
+    let text = '*Reminders!*';
+    let pendingCount: number = 0;
+    let inProgressCount: number = 0;
+
+    const result = await prisma.codeReview.groupBy({
+        by: ['status'],
+        where: {
+            status: {
+                in: ['pending', 'inprogress'],
+            },
+            slackChannelId: channel,
+        },
+        _count: {
+            status: true,
+        }
+    });
+
+    result.forEach((item) => {
```

Comment:
I propose to write this a bit shorter: 
const pendingCount = result.filter(item=> item.status === 'pending').length;
const inProgressCount = result.filter(item=> item.status === 'inprogress').length;

Reply:
This form would take twice as much cpu cycle to run compare to foreach loop; We're taking the sum of the value not the length of the resulting items. I'm going to refactor this code into two statement to be a bit more self explanatory instead.

Reply:
Refactor code in commit https://github.com/penske-media-corp/code-review-bot/pull/5/commits/6a7fe433f8050588d8ad24be3ae41eabdfe529db